manager/controllers/app/data_catalog_interface.go

```diff
@@ -16,7 +19,40 @@ import (
 // Returns:
 // - an error if happened
 // - the new asset identifier
-func (r *FybrikApplicationReconciler) RegisterAsset(catalogID string, info *app.DatasetDetails,
+func (r *FybrikApplicationReconciler) RegisterAsset(assetID string, catalogID string, info *app.DatasetDetails,
 	input *app.FybrikApplication) (string, error) {
-	return "", errors.New("unsupported feature")
+	r.Log.Trace().Msg("RegisterAsset")
+	details := datacatalog.ResourceDetails{}
+	if info.Details != nil {
+		infoDetails := info.Details
+		details.Connection = infoDetails.Connection
+		details.DataFormat = infoDetails.Format
+	}
+
+	resourceMetadata := datacatalog.ResourceMetadata{
+		Name: assetID,
+	}
+
+	request := datacatalog.CreateAssetRequest{
+		ResourceMetadata:     resourceMetadata,
+		Details:              details,
+		SecretName:           info.SecretName,
+		SecretNamespace:      info.SecretNamespace,
```

Comment:
You assume the connector can access the secret directly. This is not always the case.